common/djangoapps/student/models.py

```diff
@@ -1155,7 +1155,7 @@ def is_paid_course(self):
         """
         paid_course = CourseMode.objects.filter(Q(course_id=self.course_id) & Q(mode_slug='honor') &
                                                 (Q(expiration_datetime__isnull=True) | Q(expiration_datetime__gte=datetime.now(pytz.UTC)))).exclude(min_price=0)
-        if paid_course or self.mode == 'professional' or self.mode == CourseMode.NO_ID_PROFESSIONAL_MODE[0]:
+        if paid_course or self.mode == 'professional' or self.mode == CourseMode.NO_ID_PROFESSIONAL_MODES[0]:
```

Comment:
Once you make the change in my other comment, this should use class methods on the `CourseMode` object to determine whether course is paid or professional.

While you're in this part of the code, please replace the query on line 1157 with a call to a `CourseMode` class method.  There is no reason why other Django apps (outside of `course_modes`) should be directly querying the `CourseMode` model.